examples/passport-oauth2-login/src/authentication-strategies/oauth2-strategy.ts

```diff
@@ -71,34 +109,38 @@ export class Oauth2Authorization
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     done: (error: any, user?: any, info?: any) => void,
   ) {
-    const userProfile: UserProfile = mapFaceBookProfile(profile);
-    this.userService
-      .findOrCreateExternalUser(userProfile)
-      .then((user: User) => {
-        user.token = accessToken;
+    if (profile.emails && profile.emails.length) {
+      this.userService
+      .findOrCreateExternalUser(profile.emails[0].value, profile, accessToken)
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      .then((user: any) => {
         done(null, user);
       })
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
       .catch((err: any) => {
-        console.log(err);
         done(err);
       });
+    } else {
+      done(new Error('email-id is required in returned profile to login'));
+    }
   }
 }
 
 /**
- *
+ * map facebook profile to user profile
  * @param user
  */
 export const mapFaceBookProfile: UserProfileFactory<Profile> = function (
-  user: Profile,
+  facebookProfile: Profile,
 ): UserProfile {
+  let user = facebookProfile as UserWithToken;
   const userProfile: UserProfile = {
     [securityId]: user.id,
     profile: {
       ...user,
-      source: 'facebook',
-    },
+      _json: null,
```

Comment:
whats _json and _raw? Perhaps provide a comment. thx.

Reply:
agreed